### ISSUE TITLE
Feature Request: `AWS::StepFunctions::StateMachine` - Recognize if `Next` is missing in `Choices`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4077]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Currently cfn-lint is unable to recognize if `Next` is missing in `Choices`. The following CloudFormation Template will cause a CloudFormation error:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        StartAt: Pass
        States:
          Pass:
            Type: Pass
            Next: Choice
          Choice:
            Type: Choice
            Choices:
              - Variable: "$"
                StringMatches: "Test"
                # Next: Pass # <------------------ This is needed
            Default: Success
          Success:
            Type: Succeed
```

CloudFormation error:

```
Resource handler returned message:
"Invalid State Machine Definition: 'SCHEMA_VALIDATION_FAILED: These fields are required: [Next] at /States/Choice/Choices[0],
MISSING_TRANSITION_TARGET: Missing transition pointer at /States/Choice/Choices[0]/Next'
(Service: Sfn, Status Code: 400, Request ID: f...d)
(SDK Attempt Count: 1)"
(RequestToken: 7...9, HandlerErrorCode: InvalidRequest)
```

It would be great if cfn-lint could give an error on these.

### Describe the solution you'd like

A new/updated rule to notify the user of this error.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
